test/server/worker/simpleworker.spec.js

```diff
@@ -1742,6 +1751,64 @@ describe('Simple worker', function () {
             .nodeify(done);
     });
 
+    it('exportProjectToFile should export library when name provided', function (done) {
+        var worker = getSimpleWorker();
+
+        worker.send({command: CONSTANTS.workerCommands.initialize, gmeConfig: gmeConfig})
+            .then(function (msg) {
+                expect(msg.pid).equal(process.pid);
+                expect(msg.type).equal(CONSTANTS.msgTypes.initialized);
+
+                return worker.send({
+                    command: CONSTANTS.workerCommands.exportProjectToFile,
+                    projectId: libProjectImportResult.project.projectId,
+                    commitHash: libProjectImportResult.commitHash,
+                    libraryName: 'A'
+                });
+            })
+            .then(function (msg) {
+                expect(msg.pid).equal(process.pid);
+                expect(msg.type).equal(CONSTANTS.msgTypes.result);
+                expect(msg.error).equal(null);
+
+                expect(msg.result).not.equal(null);
+                expect(typeof msg.result).to.equal('object');
+                expect(msg.result.downloadUrl).to.include('/rest/blob/download/');
+                expect(msg.result.fileName).to.include('guest+A_');
+            })
+            .finally(restoreProcessFunctions)
+            .nodeify(done);
+    });
+
+    it('exportProjectToFile should export nested library when name provided', function (done) {
+        var worker = getSimpleWorker();
+
+        worker.send({command: CONSTANTS.workerCommands.initialize, gmeConfig: gmeConfig})
+            .then(function (msg) {
+                expect(msg.pid).equal(process.pid);
+                expect(msg.type).equal(CONSTANTS.msgTypes.initialized);
+
+                return worker.send({
+                    command: CONSTANTS.workerCommands.exportProjectToFile,
+                    projectId: libProjectImportResult.project.projectId,
+                    commitHash: libProjectImportResult.commitHash,
+                    libraryName: 'A.B'
+                });
+            })
+            .then(function (msg) {
+                expect(msg.pid).equal(process.pid);
+                expect(msg.type).equal(CONSTANTS.msgTypes.result);
+                expect(msg.error).equal(null);
+
+                expect(msg.result).not.equal(null);
+                expect(typeof msg.result).to.equal('object');
+                expect(msg.result.downloadUrl).to.include('/rest/blob/download/');
+                expect(msg.result.fileName).to.include('guest+B_');
+            })
+            .finally(restoreProcessFunctions)
+            .nodeify(done);
+    });
+
     it('should fail to exportProjectToFile when branchName does not exist.', function (done) {
         var worker = getSimpleWorker();
 
```

Comment:
any chance to add a test that would use the exported libary and run some traversal to see if everything goes well during the read of the nodes?